plugin.video.srgssr_ch_replay/resources/lib/router.py

```diff
@@ -0,0 +1,80 @@
+"""Routing in the plugin menus"""
+
+from urllib.parse import urlencode
+
+import xbmcplugin
+
+
+class Router:
+    """Router dispatching the queries on the Addon"""
+    def __init__(self, plugin):
+        self.plugin = plugin
+
+    def url(self, bu: str = "", mode: str = "", **kwargs):
+        """Constructs the plugin's URL"""
+        if not bu:
+            bu = self.plugin.bu
+        path = f"{bu}/{mode}"
+        return f"{self.plugin.ADDON_URL}/{path}?{urlencode(kwargs)}"
```

Comment:
(Major) Issue:
After a broadcast has been watched the "marked watched" ✔️ does not appear.
Probable solution: 
When I changed the URL back to the previous v3.0.3 format 
return f"{self.plugin.ADDON_URL}?bu={bu}&mode={mode}&{urlencode(kwargs)}" 
the marked as watched seems to work again. 

@joachimBurket Can you confirm that? If yes, then we need to change the URL to a stable format.

Reply:
Please have a look at this commit: https://github.com/ManBehindMooN/kodi_plugin_video_srgssr_ch_replay/pull/15/commits/46e9ccaf82be1556f5d3c67bde493985d8400c7f

Reply:
Minor issue: 
When searching TV shows or broadcasts there is an add-on/Kodi error for these two scenarios:
* Pressing "cancel" => should just return
* Pressing "OK" without any input -> should just return

Reply:
Good catch! :tada: